src/Uno.UI/UI/Xaml/Controls/Border/Border.cs

```diff
@@ -31,255 +31,260 @@
-		/// {
-		///    new Border()
-		/// }
-		/// </summary>
-		/// <param name="view"></param>
-		public
+	/// <summary>
+	/// Support for the C# collection initializer style.
+	/// Allows items to be added like this
+	/// new Border
+	/// {
+	///    new Border()
+	/// }
+	/// </summary>
+	/// <param name="view"></param>
+	public
 #if __IOS__
-			new
+		new
 #endif
-			void Add(View view)
-		{
-			Child = VisualTreeHelper.TryAdaptNative(view);
-		}
+		void Add(View view)
+	{
+		Child = VisualTreeHelper.TryAdaptNative(view);
+	}
 
-		protected override bool IsSimpleLayout => true;
+	protected override bool IsSimpleLayout => true;
 
-		private protected override Thickness GetBorderThickness() => BorderThickness;
+	private protected override Thickness GetBorderThickness() => BorderThickness;
 
 
-		#region Child DependencyProperty
+	#region Child DependencyProperty
 
-		public UIElement Child
+	public UIElement Child
+	{
+		get => (UIElement)this.GetValue(ChildProperty);
+		set => this.SetValue(ChildProperty, value);
+	}
+
+	public static DependencyProperty ChildProperty { get; } =
+		DependencyProperty.Register(
+			"Child",
+			typeof(UIElement),
+			typeof(Border),
+			new FrameworkPropertyMetadata(
+				null,
+				// Since this is a view, inheritance is handled through the visual tree, rather than via the property. We explicitly
+				// disable the property-based propagation here to support the case where the Parent property is overridden to simulate
+				// a different inheritance hierarchy, as is done for some controls with native styles.
+				FrameworkPropertyMetadataOptions.ValueDoesNotInheritDataContext,
+				(s, e) => ((Border)s)?.OnChildChanged((UIElement)e.OldValue, (UIElement)e.NewValue)
+			)
+		);
+
+	private void OnChildChanged(UIElement oldValue, UIElement newValue)
+	{
+		ReAttachChildTransitions(oldValue, newValue);
+
+		if (oldValue is not null)
 		{
-			get => (UIElement)this.GetValue(ChildProperty);
-			set => this.SetValue(ChildProperty, value);
+			RemoveChild(oldValue);
 		}
 
-		public static DependencyProperty ChildProperty { get; } =
-			DependencyProperty.Register(
-				"Child",
-				typeof(UIElement),
-				typeof(Border),
-				new FrameworkPropertyMetadata(
-					null,
-					// Since this is a view, inheritance is handled through the visual tree, rather than via the property. We explicitly
-					// disable the property-based propagation here to support the case where the Parent property is overridden to simulate
-					// a different inheritance hierarchy, as is done for some controls with native styles.
-					FrameworkPropertyMetadataOptions.ValueDoesNotInheritDataContext,
-					(s, e) => ((Border)s)?.OnChildChanged((UIElement)e.OldValue, (UIElement)e.NewValue)
-				)
-			);
-
-		private void OnChildChanged(UIElement oldValue, UIElement newValue)
+		if (newValue is not null)
 		{
-			ReAttachChildTransitions(oldValue, newValue);
-
-			OnChildChangedPartial(oldValue, newValue);
+			AddChild(newValue);
 		}
```

Comment:
I think `AddChild` internally does a null check (at least on Skia)?
If that's the case on all platforms, I think we can just call `AddChild` unconditionally here.

Reply:
Just confirmed, this is the case only on Skia and Wasm, so let's keep the null check as it's likely needed for all mobile platforms.

Reply:
The check is not uniform unfortunately. I could move the check to `VisualTreeHelper.AddChild` though (same for `RemoveChild`)